content/articles/understanding-how-cluster-marker-works/index.md

```diff
@@ -0,0 +1,240 @@
+```
+
+Changes won't affect yet on your browser, because we haven't initialized our map. To initialize, the map, set the map-view to Africa using Google Street Map. Create a script tag, copy and paste the snippet below tho the script tag.
+
+### Initialization of Map
+
+```js
+var map = L.map('my_map').setView([9.1021, 18.2812], 3);
+```
+
+The code snippet above initiates the map variable into the map using the id of `my_app`, setting the `longitude - 9.1021`, `latitude - 18.2812`, and the zoom `level - 3`. To add an OpenStreetMap we'll make use of `Google Street Map`. 
+
+Create another script tag and add the code below inside.
+
+```js
+var osm = L.tileLayer('http://{s}.google.com/vt/lyrs=m&x={x}&y={y}&z={z}', { maxZoom: 50, subdomains: ['mt0', 'mt1', 'mt2', 'mt3'] }); osm.addTo(map);
+```
+
+The script will look like this:
+
+```js
+<script type="text/javascript"> var map = L.map('my_map').setView([9.1021, 18.2812], 3); var osm = L.tileLayer('http://{s}.google.com/vt/lyrs=m&x={x}&y={y}&z={z}', { maxZoom: 50, subdomains: ['mt0', 'mt1', 'mt2', 'mt3'] }); osm.addTo(map); </script>
+```
+
+After adding the `OSM` you will see the output which will look like the image below, and still retain the zoom level we set to the map with the `longitude` and `latitude`.
+
+![image info](engineering-education/understanding-how-cluster-marker-works/map.png)
+
+### Adding Data To Map
+Any kind of data can be added to the map. but for the sake of this tutorial, we'll use the data generated from [geojson](geojson.io). Below is the image of data generated from [geojson](geojson.io).
+
+![image info](engineering-education/understanding-how-cluster-marker-works/data.png)
+
+Save and download the file generated as a GeoJson file. Create a file naming it `mapdata.js` Inside the `mapdata.js` create a variable map_data setting the value to your downloaded generated data. To create the variable, copy and paste the snippet below.
+
+```js
+const map_data = {
+
+}
+```
+
+Go to the file you just downloaded and copy all the data into the `mapdata.js` variable that is initialize 
+
+## Visualize geojson data on your map
+To render the generated data on the map, you have to link the data to the map.
+
+```sh
+<script src="./mapdata.js" type="text/javascript"></script>
+```
+
+If you check the browser, you will notice that there are no changes. To render the data on the map you have to load the geojson data using the snippet below:
+
+```js
+L.geoJSON(map_data).addTo(map);
+```
+
+i.e Load geojson data from `mapdata.js` with variable name `map_data` Your code should look like this:
+
+```js
+// Map Declearation
+var map = L.map('my_map').setView([9.1021, 18.2812], 3);
+
+var osm = L.tileLayer('http://{s}.google.com/vt/lyrs=m&x={x}&y={y}&z={z}', {
+    maxZoom: 50,
+    subdomains: ['mt0', 'mt1', 'mt2', 'mt3']
+});
+osm.addTo(map);
+
+//GeoJson Data
+L.geoJSON(map_data).addTo(map);
+```
```

Comment:
what does the above code do

Reply:
It's used for loading the map data that was initialize